### DISCUSSION TITLE
Exercise in 05_looping.ipynb

### DISCUSSION THREAD

**Discussion by whereisananya** [giswqs/geog-312#56] [Q&A] [answered]
The last question of the exercise in Looping and conditionals statements says:

5. Write a program that generates random coordinates (latitude and longitude) and prints them until a coordinate with both latitude and longitude greater than 100 is generated.

Latitude shouldn't be greater than 90 so maybe this needs to be modified.

**Comment by whereisananya** [comment]
Same in exercise 10 of lab 2 

Exercise 10: Generating and Analyzing Random Coordinates

Write a program that generates random coordinates (latitude and longitude between -180 and 180 degrees).
Use a while loop to keep generating coordinates until a pair with both latitude and longitude greater than 100 is generated.
Print each generated coordinate and the final coordinate that meets the condition.

**Reply by giswqs** [reply, marked as answer]
Thanks for checking. It has been updated.

**Reply by whereisananya** [reply]
Thanks!

**Comment by WileLee** [comment]
I know I am a bit late on these exercises (I am taking this as a personal development course online) but I was wondering how some of you may have structured Ex. 5 in this lecture. 

I have run through a few versions that I've had varying levels of success with. Below is my most recent version and I feel like I'm moving backwards in success. With that, would anyone be willing to share their work and thought process to get there? 
I appreciate the collaboration in advance!

```
# Step 5 - Random coordinate generator (Until both Lat & Lon are > 50 degrees)

import random
random_coord = []
random_coord = [(random.uniform(49, -90), random.uniform(49, -180))]
gen_count = 1

print(f"Starting random generator for coordinates...")
print(f"Random generation number {gen_count}: Lat: {lat:.6f}, Lon: {lon:.6f}")


for lat, lon in random_coord:
    if lat < 50:
        if lon < 50:
            gen_count += 1
            lat = random.uniform(90, -90)
            lon = random.uniform(180, -180)
            print(f"Random generation number {gen_count}: Lat: {lat:.6f}, Lon: {lon:.6f}")
        else:
            break

print(f"The list generated {gen_count} times: ({lat:.6f}, {lon:.6f}).")

```

**Reply by WileLee** [reply]
I did finally lean on some assistance to clean this up. I was way overthinking towards the end. I was close with my original attempts that I didn't share. 

I was viewing things through nests far more than I should have. I don't think we covered leveraging an 'or' within the while loop in lecture but that's what I was trying to achieve with an 'and' in one of my 'for' loops. 

My solution is below:

```
# Step 5 - Random coordinate generator (Until both Lat & Lon are > 50 degrees)

import random
random_coord = []
random_coord = [(random.uniform(-90, 90), random.uniform(-180, 180))]
gen_count = 1
lat, lon = random_coord[0]

print(f"Starting random generator for coordinates...\n.\n.\n.\n.")
print(f"The list generated {gen_count} times: ({lat:.6f}, {lon:.6f})")


while lat <= 50 or lon <=50:
    gen_count += 1
    lat = random.uniform(-90, 90)
    lon = random.uniform(-180, 180)
    print(f"The list generated {gen_count} times: ({lat:.6f}, {lon:.6f})")

print(f"The list generated {gen_count} times to create coords greater than (50N, 50E): ({lat:.6f}, {lon:.6f})")
```
